core/Scheduler/Scheduler.php

```diff
@@ -147,6 +152,33 @@ public function run()
                     $readFromOption = true;
                     $message = $this->executeTask($task);
 
+                    // Task has thrown an exception and should be scheduled for a retry
+                    if ($this->scheduleRetry) {
+
+                        if($this->timetable->getRetryCount($task->getName()) == 3) {
+
+                            // Task has already been retried three times, give up
+                            $this->timetable->clearRetryCount($task->getName());
+
+                            $this->logger->warning("Scheduler: '{task}' has already been retried three times, giving up",
+                                ['task' => $task->getName()]);
+
+                        } else {
+
+                            $readFromOption = true;
+                            $rescheduledDate = $this->timetable->rescheduleTaskAndRunInOneHour($task);
```

Comment:
very random thought, not sure it's an issue. What if the task would run in one hour again anyway because it's running hourly? If I understand it right we would still only execute it once in the next hour and seems all good 👍  and probably in that case we don't need to do anything special.

Reply:
Yes, if a task was scheduled hourly then the retry schedule would be the same as it's next execution so there shouldn't be any difference.